File: libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/network/NetworkTransport.java
package com.apollographql.apollo3.runtime.java.network;

import com.apollographql.apollo3.api.ApolloRequest;
import com.apollographql.apollo3.api.Operation;
import com.apollographql.apollo3.runtime.java.ApolloCallback;
import com.apollographql.apollo3.runtime.java.ApolloDisposable;
import org.jetbrains.annotations.NotNull;

public interface NetworkTransport {
  <D extends Operation.Data> void execute(@NotNull ApolloRequest<D> request, @NotNull ApolloCallback<D> callback, ApolloDisposable disposable);

  void dispose();
}

File: apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo3/network/NetworkTransport.kt
package com.apollographql.apollo3.network

import com.apollographql.apollo3.api.Operation
import com.apollographql.apollo3.api.ApolloRequest
import com.apollographql.apollo3.api.ApolloResponse
import kotlinx.coroutines.flow.Flow

interface NetworkTransport {

  fun <D : Operation.Data> execute(
      request: ApolloRequest<D>,
  ): Flow<ApolloResponse<D>>

  fun dispose()
}
